### PR TITLE
Implemented removing the ingredients needed to make a specified recipe from inventory

### LINKED ISSUE #59: As a user I can remove all the ingredients needed to make a specified recipe from my inventory
...so that I do not have to manually remove the required quantity of each individual item required to make the recipe

### PR DESCRIPTION
**Pull request by TJW0911**
resolves #59

### CODE REVIEW

File: src/main/java/seedu/mealcompanion/command/misc/MakeCommand.java [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+            IngredientList fridgeIngredients = mealCompanionSession.getIngredients();
+            if (canMakeRecipe(recipe, fridgeIngredients)) {
+                makeRecipe(mealCompanionSession, recipe);
+            } else {
+                throw new MealCompanionException("Ingredients in inventory is insufficient");
+            }
+        } catch (Exception e) {
+            System.out.println(e);
+        }
+    }
```
Comment (EthanYidong) [inline src/main/java/seedu/mealcompanion/command/misc/MakeCommand.java:146]: Could you include some tests for this function?

Comment (TJW0911) [reply src/main/java/seedu/mealcompanion/command/misc/MakeCommand.java:156]: Added

File: src/main/java/seedu/mealcompanion/command/misc/MakeCommand.java [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+     * Removes a specified quantity of an ingredient from the ingredient list
+     *
+     * @param mealCompanionSession the MealCompanionSession containing the list of ingredients
+     * @param quantity the quantity of ingredient to be removed
+     * @param name the name of the ingredient to be removed
+     */
+
+    private void removeIngredient(MealCompanionSession mealCompanionSession, Double quantity, String name) {
+        int indexOfExistingIngredient = findIngredient(mealCompanionSession, name);
+        double fridgeQuantity = mealCompanionSession.getIngredients().get(indexOfExistingIngredient).getQuantity();
```
Comment (ngyida) [inline src/main/java/seedu/mealcompanion/command/misc/MakeCommand.java:103]: is the case for index = -1 handled?

File: src/main/java/seedu/mealcompanion/command/misc/MakeCommand.java [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+        assert fridgeQuantity >= quantity: "fridgeQuantity should be more than quantity to be removed";
```
[the remark this answers is not in the document]
Comment (TJW0911) [reply src/main/java/seedu/mealcompanion/command/misc/MakeCommand.java:104]: From the hasEnoughIngredient() method we have already implicitly determined that the users ingredient list definitely has sufficient ingredients for the recipe, hence the assert is used

File: src/main/java/seedu/mealcompanion/command/misc/MakeCommand.java (hunk 1 above)
Comment (Jjzeng123) [reply src/main/java/seedu/mealcompanion/command/misc/MakeCommand.java:156]: looks good

**Pull request merged**